wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/FunctionEmitter.scala

```diff
@@ -246,6 +270,36 @@ object FunctionEmitter {
   }
 
   private type Env = Map[LocalName, VarStorage]
+
+  object SWasmTrees {
+    abstract sealed class SWasmTree {
```

Comment:
If you want custom trees that integrate well within IR trees, you can use `ir.Trees.Transient` nodes. They carry an arbitrary payload that implements `Transient.Value`.